### PR TITLE
CIF-1327 - Show a "Create configuration folder" action in the CIF Configuration console

### PR DESCRIPTION
**Pull request by LSantha**
* implemented create configuration folder action
 * grouped create actions in create pull-down
 * adjusted display logic for displaying empty configuration folders
 * adjusted delete action for removing empty configuration folders
 * clarified render conditions
 * updated unit tests

## Description

The "Create configuration folder" action should be available in the CIF Configuration console to avoid switching between consoles.

Ideally the new configuration folder would already have the settings/cloudservices subtree created instead of the user explicitly selecting "Cloud Services" from the dialog.

## Related Issue

CIF-1327

## How Has This Been Tested?

Manually, JUnit.

## Screenshots (if appropriate):

screenshot1

screenshot2

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes and the overall coverage did not decrease.
- [X] All unit tests pass on CircleCi.
- [X] I ran all tests locally and they pass.

### CODE REVIEW

File: content/cif-connector/src/main/content/jcr_root/libs/commerce/gui/clientlibs/configbrowser/js/renderconditions/renderconditions.js [hunk 1]
```diff
@@ -11,43 +11,58 @@
...
-        let relAction;
+        let actionRels;
         if (activeItemMeta) {
-            relAction = activeItemMeta.dataset['foundationCollectionQuickactionsRel'];
+            actionRels = activeItemMeta.dataset['foundationCollectionQuickactionsRel'];
         }
+
+        const createPullDown = document.querySelector(CREATE_PULLDOWN_SELECTOR);
+        hide(createPullDown);
+        if (actionRels && createPullDown && actionRels.indexOf(CREATE_PULLDOWN_REL) >= 0) show(createPullDown);
```
Comment (dplaton) [inline content/cif-connector/src/main/content/jcr_root/libs/commerce/gui/clientlibs/configbrowser/js/renderconditions/renderconditions.js]: Missing braces around `show`

File: content/cif-connector/src/main/content/jcr_root/libs/commerce/gui/clientlibs/configbrowser/js/renderconditions/renderconditions.js [hunk 2]
```diff
@@ -11,43 +11,58 @@
...
+
+        const createFolderButton = document.querySelector(CREATE_FOLDER_SELECTOR);
+        hide(createFolderButton);
+        if (actionRels && createFolderButton && actionRels.indexOf(CREATE_FOLDER_REL) >= 0) show(createFolderButton);
```
Comment (dplaton) [inline content/cif-connector/src/main/content/jcr_root/libs/commerce/gui/clientlibs/configbrowser/js/renderconditions/renderconditions.js]: Same as above

File: bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/gui/components/configuration/ConfigurationColumnViewItem.java [hunk 3]
```diff
@@ -64,34 +67,66 @@ public String getTitle() {
...
-                JcrResourceConstants.NT_SLING_ORDERED_FOLDER))
+            .startsWith(Constants.CONF_ROOT) && (resource.isResourceType(JcrConstants.NT_FOLDER) ||
+                resource.isResourceType(JcrResourceConstants.NT_SLING_FOLDER) ||
+                resource.isResourceType(JcrResourceConstants.NT_SLING_ORDERED_FOLDER))
             && resource
                 .getChild(Constants.CONF_CONTAINER_BUCKET_NAME + "/" + Constants.CLOUDCONFIGS_BUCKET_NAME) != null);
     }
+
+    private boolean isSafeToDelete() {
+        return isCommerceBucket() || resource.getPath().startsWith(Constants.CONF_ROOT)
```
Comment (dplaton) [inline bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/gui/components/configuration/ConfigurationColumnViewItem.java]: Can we optimize this condition? Split similar conditions into boolean methods with meaningful names is one option.

The way it is now is barely readable.

Maybe I'm nitpicking, but I think `StreamSupport.stream(resource.getChild(Constants.CONF_CONTAINER_BUCKET_NAME).getChildren().spliterator(), false).count()` can be wrapped in a method to give it a more descriptive name.

**Review by dplaton — CHANGES_REQUESTED** [review]
The implementation looks ok, I just had some comments regarding the code style.

File: bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/gui/components/configuration/ConfigurationColumnViewItem.java (hunk 3 above)
Comment (LSantha) [reply bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/gui/components/configuration/ConfigurationColumnViewItem.java]: Yeah, those conditions have gotten a bit wild..., although the cases they are trying to catch are quite weird too.

File: content/cif-connector/src/main/content/jcr_root/libs/commerce/gui/clientlibs/configbrowser/js/renderconditions/renderconditions.js (hunk 1 above)
Comment (LSantha) [reply content/cif-connector/src/main/content/jcr_root/libs/commerce/gui/clientlibs/configbrowser/js/renderconditions/renderconditions.js]: @dplaton, that is just following what you wrote ... :-) But never mind, I can "fix" it.

**Review by dplaton — APPROVED** [review]

**Pull request merged**